Added navigation in patet-page

daancs opened the pull request:
There is now a "scroll to top" button when you scroll down and now there are navbuttons on the side to more quickly get to a chosen year

daancs committed:
added just in time css and side-navigation for patet

daancs committed:
fixed patetnav to transition year and styling

daancs committed:
added new navBall component

daancs committed:
uses NavBall component for patet navigation now

daancs committed:
Merge branch 'dev' into feature-scrolltotop

daancs committed:
just fixed variable name for allcommittees instead of allnollkit

drakejacob commented on tailwind.config.js:
```diff
@@ -20,4 +20,11 @@ module.exports = {
     },
   },
   plugins: [],
+  mode: 'jit',
```
Do we need just in time? Or was this just a test which is no longer necessary?

daancs replied:
I don't know if its needed but I haven't tested without it, might break. Also, jit compilation is pretty nice but idk

daancs replied:
it works without so Im removing it

drakejacob commented on styles/globals.css:
```diff
@@ -47,6 +47,14 @@ div:hover > .downarrow {
   opacity: 90%;
 }
 
+.opacitykid:hover > .numberop {
```
I do not like the name opacitykid, even though it's quite funny
numberop is not descriptive either. Rename or add a comment, i suggest giving it a better name.
There is no clue it is related to the floor-selection in the pateter page if you're only looking at the CSS-file.

drakejacob commented on pages/pateter.tsx:
```diff
@@ -97,6 +116,17 @@ const Pateter: NextPage<PateterProps> = ({allCommittees}) => {
       </div>
+      
+      <div className="fixed flex flex-col items-center right-4 self-center top-1/4">
+          <NavBall index={0} scrollTo={() => scrollTo(0)} currentPage={currentPage} committeeyr={"Top"} ></NavBall>
```
committeeyr? what does that mean? if it means 'year' then write committeeyear instead?

drakejacob commented on components/NavBall.tsx:
```diff
@@ -0,0 +1,13 @@
+import { Committee } from "@prisma/client";
+import { argv0 } from "process";
```
Is this import necessary?

daancs replied:
I thought I already deleted that lmao

drakejacob commented on components/NavBall.tsx:
```diff
@@ -0,0 +1,13 @@
+    )
+};
+
+export default NavBall;
```
There is a decision to made if a component deserves its own file in the components directory, or if it's just another function inside the same file.
I would maybe not put the navball in components, and instead just put it in pateter, since I don't think it's to be used anywhere else and is somewhat taylor-made for that page.

daancs committed:
fixed allcommittees so its patet only, also formatted document

drakejacob reviewed (commented):
Good addition, and well implemented
Some changes I think should be adressed are commented

Going to add an issue regarding button styling, the scroll to top button works great, but needs to fit in to the site's design language, which we have yet to fully establish.

Also, it would be cool if there was some transition applied to the opacity of the navball when you move through the years so it doesn't just appear. Maybe there already is some, if so it is hard to notice it.

daancs commented:
closed until further fixes

drakejacob commented:
> closed until further fixes

I don't know, I think closed pull requests are cancelled ideas?
And unfinished should remain open but not be merged, but im not sure.
Would be nice if there was a middle ground like "paused" pull request.
But im not sure what the convention is

daancs commented:
> > closed until further fixes
> 
> I don't know, I think closed pull requests are cancelled ideas? And unfinished should remain open but not be merged, but im not sure. Would be nice if there was a middle ground like "paused" pull request.

yuh, idk, I feel like an open pull request = finished code that I want to merge into prod

daancs committed:
removed jit css compilation

daancs commented:
> Also, it would be cool if there was some transition applied to the opacity of the navball when you move through the years so it doesn't just appear. Maybe there already is some, if so it is hard to notice it.

I added some now but you don't really notice it at all since the ball gets resized. This can maybe be changed to look better

Edit: nvm, looks a lot better with right adjustments

daancs committed:
added opacity transitioning to balls
changed variablename

daancs committed:
changed variable

daancs commented:
- [x] Add opacity transitioning to balls
- [x] Remove JIT compiling
- [x] Change variable names for committee year
- [x] Remove unused imports
- [x] Move navball component into pateter.tsx
- [x] Better descriptive names for css-classes

daancs committed:
updated css-class names to more intuitive names

daancs committed:
moved navball component to pateter.tsx
also changed some QoL stuff such as pointer when hovering and opacity for ball changes when also hovering text etc.

daancs committed:
fixed styling for topbutton
